packages/react-performance/src/test/performance-effect-server.test.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import {mount} from '@shopify/react-testing';
+import {mockPerformance} from './utilities';
+import {
+  usePerformanceEffect,
+  PerformanceEffectCallback,
+  PerformanceContext,
+} from '..';
+
+jest.mock('@shopify/performance', () => {
+  return {
+    ...require.requireActual('@shopify/performance'),
+    Performance: jest.fn(),
+  };
+});
+
+jest.mock('../utilities', () => ({
+  isServer: () => true,
+}));
```

Comment:
If this was the only reason for the function, you can definitely avoid this by setting `jest-env` :)

Reply:
it is the only reason 😅 
I tried doing it otherwise with environment setting for the whole file then `react-testing/mount` would fail because in the test wrapper uses `document`

Reply:
IMO its either we do a strange workaround like this or skip the test all together

Reply:
Yeah we can't use `react-testing`, but it doesn't really make sense to do so anyways, because we are actually testing server behaviour, and react-testing is really react-dom-testing. I think it would be appropriate to use react's server API here (`renderToString`) instead, should still be able to keep the rest of the change relatively unchanged.